### PR TITLE
5 feature 1 user registration & log in

### PR DESCRIPTION
**Pull request by CC-3636**
All classes / interfaces in Entity layer: User, StudentUser, InstructorUser, UserWhisperer, internalUserList

### CODE REVIEW

**Review by celinelumqr — DISMISSED** [review]
attributes necessary for Calendar feature are contained in StudentUser.

**Review by jltng — APPROVED** [review]
attributes necessary for course enrolment + course creation are present

File: src/main/java/entities/User.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package entities;
+
+// Entity
+public interface User {
+
```
Comment (yyutiw) [inline src/main/java/entities/User.java:5]: add Javadoc comments once you start implementing :)

File: src/main/java/entities/internalUserList.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package entities;
+
+import java.util.ArrayList;
+
+public class internalUserList {
+
```
Comment (yyutiw) [inline src/main/java/entities/internalUserList.java:6]: not 100% sure about this but wouldn't you need a getter/setter to access and modify this list?

**Review by yyutiw — COMMENTED** [review]
InternalUserList may need additional methods to be accessed, but should be ok to merge as long as no other feature relies heavily on being able to access it.

Otherwise, the attributes necessary for Task creation and maintenance are all present.

File: src/main/java/entities/internalUserList.java (hunk 2 above)
Comment (CC-3636) [reply src/main/java/entities/internalUserList.java:6]: Yea I think so--see User Story

**Review by alyson647 — DISMISSED** [review]
Attributes necessary for scheduling collaborative tasks are contained in StudentUser.

**Review by TiareMar — COMMENTED** [review]
I'm pretty sure we don't want your .gitignore, .idea and gradle changes merged to the main branch for all of us - just the code in src/main/java/entities. Is there a way to remove them?

Besides that, your code looks good to me.

**Comment by alyson647** [comment]
Sorry, I approved the pull request but I just had one comment. As you commented in the code, for workingHours, I think we may need to change the type. I might be wrong but I think DateTimeFormatter is just for formatting the date in an easy to read way? I think I'm using LocalDateTime in my code but I can change it if you decide to change the type to something else. Also, it's a list of times but how would we differentiate between the start time and end time of a certain time they blocked off, given that they can block off multiple times? As well, will all the times blocked off in workingHours be recurring?

**Comment by CC-3636** [comment]
@alyson647 I switched the type to LocalDateTime (have not pushed this change yet). I see the issue with differentiating between start and end time. What if we instead had a list of tuples (or lists) which contain start and end time? Also, I think it would be simpler to have all the times blocked off in workingHours recurring, but I don't mind if they're not (it doesn't make a huge difference to my code). What do you think?

**Comment by alyson647** [comment]
@CC-3636 I think it makes sense having a list of tuples (or lists) which contain start and end time. I also agree, with having the workingHours recurring since it would be simpler.

**Review by alyson647 — APPROVED** [review]
Attributes needed for scheduling collaborative tasks are present.

**Comment by TiareMar** [comment]
Small adjustment: desiredGrades map should be a string to double map. The string is probably going to be course ids but you can also make it course name if you think that's better

**Pull request merged**